### DISCUSSION TITLE
How to get UUID from parameters?

### DISCUSSION THREAD

**Discussion by Genebson** [cypress-io/cypress#26603] [Questions and Help]
Hey :wave:,

I have a button that when you click it(let's suppose that you're on `/order`), redirects you to a new page with a generated UUID passed through URL parameters (`/order/sl435jhsdfhkj`). 

![image](https://user-images.githubusercontent.com/65625400/234737513-fe851ea8-de44-40e2-8b7e-e30e3a9d2b1b.png)

I've tried something like this, but it doesn't work:
```
  it("Should order", () => {
    cy.visit("/order");
    cy.get(".Step1Wrapper").should("be.visible");
    cy.get(".OrderItem")
      .contains("Order now")
      .click()
      .then(() => {
        cy.wait("@order");
        cy.url().then((url) => {
          console.log("url", url);
        });
      });
  });
  ```

Is it possible?

Thank you.

**Comment by portyu9** [comment]
Hi @Genebson 

In your example the UUID is not a query parameter; it is a **path segment**:

```text
/order/sl435jhsdfhkj
       ^^^^^^^^^^^^^^
```

After waiting for the order request/navigation, get the `pathname` and extract the final segment:

```js
cy.wait('@order')

cy.location('pathname')
  .should('match', /^\/order\/[^/]+$/)
  .then((pathname) => {
    const uuid = pathname.split('/').pop()

    cy.log(`Order UUID: ${uuid}`)
    // use uuid here
  })
```

`cy.location()` is preferable to manually reading `window.location`, and its assertions are retried, so the test can wait for the redirect to complete.

If the `@order` API response itself returns the newly-created order ID, that is even better: take the ID from `cy.wait('@order').its('response.body...')` and then assert that the browser navigated to `/order/<id>`. That avoids parsing the URL to rediscover data the API already returned.

Please mark this answer as accepted if it helped, thank you.
